### PR TITLE
Prepare the package to be installed via NPM

### PR DESCRIPTION
**Pull request by lzrski**
There are three main build targets:

  - dist/index.js

    A ES6 / CommonJS module that can be required or imported in projects 
that support modules.

  - dist/salary-calculator.js

    A script exposes global SalaryCalculator variable. Can be simply 
referenced in an HTML script tag. See website/custom.html

  - dist/index.html

    an HTML file that can be directly opened in the browser. Keep in 
mind that to work it needs other files in dist/ directory. If copying to 
another project, copy entire directory. See website/Makefile for example 
of that.

There is also a website/ directory containing a sample integration with 
outside projects. Later we can use it for:

  - Publishing stand alone website for Salary Calculator
  - Automated integration testing

Caveats:

  - I had to tweak Makefile a bit. ~~Most notably the .installed file is 
now created before the npm ci is run. That's to avoid endless look (make 
-> npm ci -> make ...). I'm not happy about it. One problem is that if 
the npm ci command fails or is terminated, then make wrongly assumes 
that dependencies are installed. Another is that test, link and build 
are running twice after change in dependencies. For npm install there is 
--ignore-scripts flag that would be appropriate here, but ther 
apparently is no equivalent for npm ci.~~ (fixed in second commit)

  - The dist/index.html file assumes that script is at 
/salary-calculator.js (absolute path).  There may be a way to tweak 
parcel to make it relative.

### CODE REVIEW

**Comment by zupo** [comment]
> Most notably the .installed file is now created before the npm ci is run.

This is quite a blocker.

This PR seems a bit too much of code & configuration for the "simple" task of publishing a compiled elm.js file.

**Comment by lzrski** [comment]
The `.installed` thing is a bummer for sure. Maybe we can find a better way together. I'm really clumsy with make. 

> [...] a bit too much of code & configuration [...]

Really? I count 9 lines of code more than before: 

- 4 lines added to `Makefile`,
- 3 to `index.html`,
- 1 to `src/index.js`
- 1 to `src/salary-calculator.js`

Everything that is in `website/` directory is just an example. You can delete this directory and it will still work the same. 

The benefit is that you can just install it in minisites the standard way JS dependencies are installed (NPM or Yarn). Then you get semantic versioning, modules and script tags working. This is how everybody in front-end works. I really feel that we should talk about this before you make your mind.

**Comment by lzrski** [comment]
The `.installed` problem is addressed.

File: Makefile [hunk 1]
```diff
@@ -27,13 +27,29 @@ tests: .installed
...
+
+# Publish a pre-release version to NPM
+.PHONY: publish
+publish: release = $(shell date +%Y-%m-%dT%H-%M-%S)
+publish: dist
+	@npm version \
+		--no-git-tag-version \
+		prerelease \
+		--preid $(release)
+	npm publish
```
Comment (zupo) [inline Makefile]: prepend `@`

File: website/Makefile [hunk 2]
```diff
@@ -0,0 +1,41 @@
+.PHONY: all
```
Comment (zupo) [inline website/Makefile]: Let's take `website/` out, it does not add a lot of value. I'll use niteo.co/salary-calculator as a demo.

**Pull request merged**